runtime/src/bank/partitioned_epoch_rewards/distribution.rs

```diff
@@ -150,17 +169,49 @@ impl Bank {
         })
     }
 
-    /// store stake rewards in partition
-    /// return the sum of all the stored rewards
+    /// Store stake rewards in partition
+    /// Returns DistributionStorageResults containing the sum of all the rewards
+    /// stored, the sum of all rewards burned, and the updated StakeRewards.
+    /// Because stake accounts are checked in calculation, and further state
+    /// mutation prevents by stake-program restrictions, there should never be
+    /// rewards burned.
     ///
-    /// Note: even if staker's reward is 0, the stake account still needs to be stored because
-    /// credits observed has changed
-    fn store_stake_accounts_in_partition(&self, stake_rewards: &[StakeReward]) -> u64 {
-        self.store_accounts((self.slot(), stake_rewards));
-        stake_rewards
-            .iter()
-            .map(|stake_reward| stake_reward.stake_reward_info.lamports)
-            .sum::<i64>() as u64
+    /// Note: even if staker's reward is 0, the stake account still needs to be
+    /// stored because credits observed has changed
```

Comment:
nit: I know this comment predates your work, but I don't think it's correct.

If a stake account has no rewards, it's because the delegation is too small to accrue rewards, or the commission split makes it that either the staker or the validator gets no lamports, or the validator didn't get enough credits during the epoch. In those cases, we do not update credits observed.

I wonder if it's worth skipping those accounts entirely in some future work :thinking: on the flipside, it's probably a very small number of accounts

Reply:
Yes, agreed on these points. I'll take a look and deal with (at least the comment) in a separate PR. Sound okay?